lessons/intro_and_strings/strings.py

```diff
@@ -100,45 +129,115 @@
 # A. Below is declared a variable called my_letters. Using the 'find' method, 
 #    write code that will find the first index of the letter 'a' in my_letters.
 my_letters = 'abcdefghabalmnoap'
+
 # PUT THE ANSWER TO A HERE
+my_letters_a = my_letters.find("a")
+print("in", my_letters, ", a first appears at index", my_letters_a)
+
 # B. Using the variable my_letters again and the find method again, find the first instance
 #    of the letter 'a' from index 2 onward.
 # PUT THE ANSWER TO B HERE
+my_letters_a_2 = my_letters.find("a", 2)
+print("after index 2, a first appears at index", my_letters_a_2)
+
 # C. Using the variable my_letters again and the find method again, find the first instance
 #   of the letter 'a' from index 11 through index 16. 
 # PUT THE ANSWER TO C HERE
+my_letters_a_11_16 = my_letters.find("a", 11, 16)
+print("inclusively (i think, WRONGLY IT TURNS OUT) between indexes 11 and 16, a first appears at index", my_letters_a_11_16)
+
+
 # D. Using the variable my_letters and the find method, try to find the letter 'z' which is not
 #    in the string. What do you think will happen?
 # PUT THE ANSWER TO D HERE
+# MH:
+#     the docs.python link tells me it will return "-1" if sub is not found
+my_letters_z = my_letters.find("z")
+print(my_letters_z)
+# MH:
+#     seems like it would be nicer to use an if?
+#     and set it if the find returns an interger<0
+#     then it prints "that letter is not found".
+
 # E. There is a letter 'a' at index 10 in my_letters. What do you think will happen if you try
 #    to find the letter 'a' at starting index 9 and ending index 10? Test it.
 # PUT THE ANSWER TO E HERE
+# MH:
+#     i think the search is inclusive (see 3.C), so i think it will find a at 10...
+my_letters_a_9_10 = my_letters.find("a", 9, 10)
+print(my_letters_a_9_10)
```

Comment:
So str.find is inclusive for the starting index and exclusive in the ending index.